events/api.py

```diff
@@ -147,6 +147,38 @@ def get_authenticated_data_source_and_publisher(request):
     return data_source, publisher
 
 
+def get_publisher_query(publisher):
+    """Get query for publisher (Organization)
+
+    Some organizations can be replaced by a new organization.
+    We need to return objects that reference on replaced
+    organization when querying the new organization, and vice
+    versa.
+
+    :param publisher: a or a list of filtering organizations
+    :type publisher: str, Organization, list
+    :return: the query that check both replaced and new organization
+    """
+    if isinstance(publisher, list):
+        q = Q(
+            publisher__in=publisher,
+        ) | Q(
+            publisher__replaced_by__in=publisher,
```

Comment:
What will happen if the replaced organization is replaced?

This probably does not happen at the moment (and I don't know if it ever will), but what if we have a chain of organizations that replace one another? I believe we should display the objects of all the previous organizations there.

Reply:
Good point, I think it should be possible to have multiple mptt tree (and multiple tree manager) in a model, I think it will require changing some existing code how we use the mptt methods (e.g. providing `tree_id` for the methods). But not sure if it's necessary to introduce it now, maybe better we introduce it in the time when we need it as there's possibility that we ain't gonna need it.

Reply:
Another option would to recursively query the parents for each provided publisher. This will not affect the code in other places, but will sacrifice the performance quite a bit.

Reply:
Hm. Yeah, thinking about it again, perhaps it is good that we do not allow downright ancient organizations to be used in queries. If there is a new replacement, I think the users should be aware of that or at least the previous replacement by that point :)